runtime/src/main/java/org/corfudb/runtime/view/stream/BackpointerStreamView.java

```diff
@@ -268,7 +301,52 @@ protected boolean fillReadQueue(final long maxGlobal,
 
         }
 
+        log.debug("Read_Fill_Queue[{}] Filled CP queue with {}", this, context.readCpQueue);
         log.debug("Read_Fill_Queue[{}] Filled queue with {}", this, context.readQueue);
-        return !context.readQueue.isEmpty();
+        return ! context.readCpQueue.isEmpty() || !context.readQueue.isEmpty();
+    }
+
+    private void examineCheckpointRecord(final QueuedStreamContext context,
+                                         ILogData currentEntry, CheckpointEntry cpEntry,
+                                         boolean considerCheckpoint, long currentRead) {
+        if (context.checkpointSuccessID == null &&
+                cpEntry.getCpType().equals(CheckpointEntry.CheckpointEntryType.END)) {
+            log.trace("Checkpoint: address {} found END, id {} author {}",
+                    currentRead, cpEntry.getCheckpointID(), cpEntry.getCheckpointAuthorID());
+            if (considerCheckpoint) {
+                context.checkpointSuccessID = cpEntry.getCheckpointID();
+                context.checkpointSuccessNumEntries = 1L;
+                context.checkpointSuccessEstBytes = (long) currentEntry.getSizeEstimate();
```

Comment:
Note that getSizeEstimate will return 1, once the entry has be deserialized, so you'll be reporting incorrect checkpoint size numbers.